contracts/contracts/vault/VaultCore.sol

```diff
@@ -565,6 +564,34 @@ contract VaultCore is VaultInitializer {
                     Pricing
     ****************************************/
 
+    /**
+     * @notice The value (USD or ETH) of the collateral assets received from
+     * redeeming 1 Origin Token (OUSD or OETH) from the Vault.
+     * This is the minimum price for the OToken. A better price is usually achieved by
+     * swapping the OToken on the Curve pool used for Automated Market Operations (AMO).
+     * For OETH, that's the Curve OETH/ETH pool.
+     * For OUSD, that's the Curve OUSD/3Crv pool.
+     * @param price the price to 18 decimals.
+     */
+    function floorPrice() external view virtual returns (uint256 price) {
+        // Get the assets for redeeming 1 OETH
+        // This has already had the redeem fee applied
+        uint256[] memory redeemAssets = _calculateRedeemOutputs(1e18);
+
+        // For each of the redeemed assets
+        for (uint256 i = 0; i < redeemAssets.length; ++i) {
+            // Sum the value of the vault asset = asset amount * oracle price
+            // For OUSD's USDC and USDT assets that are to 6 decimals, the oracle
+            // price is to 18 decimals, so we do not need to scale them up to 18 decimals
+            price +=
+                redeemAssets[i] *
+                IOracle(priceProvider).price(allAssets[i]);
```

Comment:
Leaving a note here so it is not forgotten:  we query oracle price per asset in this line above and also in [_toUnitPrice](https://github.com/OriginProtocol/origin-dollar/blob/master/contracts/contracts/vault/VaultCore.sol#L668) function withing the `_calculateRedeemOutputs `